src/main/java/fruitshop/strategy/BalanceOperation.java

```diff
@@ -0,0 +1,18 @@
+package fruitshop.strategy;
+
+import fruitshop.db.Storage;
+import fruitshop.model.FruitTransaction;
+
+public class BalanceOperation implements OperationHandler {
+    private final Storage storage = new Storage();
+
+    @Override
+    public void handler(FruitTransaction fruitTransaction) {
+        if (fruitTransaction.getQuantity() >= 0) {
+            storage.put(fruitTransaction.getFruit(), fruitTransaction.getQuantity());
+        } else {
+            throw new RuntimeException("negative balance cannot be recorded at "
+                    + BalanceOperation.class);
```

Comment:
The exception message should include more context about the error, such as the actual value that caused the exception and possibly the transaction details. This will make debugging easier if an error occurs.